packages/client/src/runtime/core/model/UserArgs.ts

```diff
@@ -1,8 +1,4 @@
 /**
  * Input that flows from the user into the Client.
  */
-export type UserArgs = {
-  [K in string]: UserArgsProp | UserArgsProp[]
-}
-
-type UserArgsProp = UserArgs | string | number | boolean | bigint | null | undefined
+export type UserArgs = any
```

Comment:
We can now also change the `args` type in `QueryMiddlewareParams` (where `InternalRequestParams` inherits it from) to explicit `UserArgs` that `argsMapper` expects. The previous too narrow type would somewhat conflict with how args were used in `runCommandRaw` method, but now might be a good idea to use an explicit type even if it's aliased to `any` so if we have a good reason to narrow it down, we can do it in one place and see the effects of doing so.

Reply:
I tried hard to keep it type safe but failed because of variance issues, and as such `unknown` isn't a good fit either. I left a comment about it.

Reply:
I applied your suggestion, and also moved `UserArgs` to another folder.